lua/gen/init.lua

```diff
@@ -113,98 +196,203 @@ M.exec = function(options)
     local prompt = opts.prompt
 
     if type(prompt) == "function" then
-        prompt = prompt({content = content, filetype = vim.bo.filetype})
+        prompt = prompt({ content = content, filetype = vim.bo.filetype })
     end
 
-    prompt = vim.fn.shellescape(substitute_placeholders(prompt))
+    prompt = substitute_placeholders(prompt)
     local extractor = substitute_placeholders(opts.extract)
-    local cmd = opts.command
+
     prompt = string.gsub(prompt, "%%", "%%%%")
-    cmd = string.gsub(cmd, "%$prompt", prompt)
-    cmd = string.gsub(cmd, "%$model", opts.model)
-    if opts.container ~= nil then
-        cmd = string.gsub(cmd, "%$container", opts.container)
-    end
-    if result_buffer then vim.cmd('bd' .. result_buffer) end
-    local win_opts = vim.tbl_deep_extend('force', get_window_options(),
-                                         opts.win_config)
-    result_buffer = vim.api.nvim_create_buf(false, true)
-    vim.api.nvim_buf_set_option(result_buffer, 'filetype', 'markdown')
-
-    local float_win = vim.api.nvim_open_win(result_buffer, true, win_opts)
-
-    local result_string = ''
-    local job_id
-    vim.api.nvim_buf_call(result_buffer, function()
-        vim.cmd('set syntax=on')
-        vim.fn.termopen(cmd, {
-            on_exit = function(a, b)
-                if b == 0 and opts.replace then
-                    local lines = trim_table(
-                                      vim.api.nvim_buf_get_lines(result_buffer,
-                                                                 0, -1, false))
-                    local text = table.concat(lines, '\n')
-                    if extractor then
-                        local extracted = text:match(extractor)
-                        if not extracted then
-                            vim.cmd('bd ' .. result_buffer)
-                            return
+
+    M.result_string = ""
+    local bodyData = {
+        model = opts.model,
+        prompt = prompt,
+        stream = true,
+    }
+
+    if M.context then
+        bodyData.context = M.context
+    end
+
+    local json = vim.fn.json_encode(bodyData)
+    local cmd = "curl --silent --no-buffer -X POST "
```

Comment:
Can we also support `M.command`?

Reply:
(otherwise it would be a breaking change)

Reply:
The biggest problem there is that the container that hosts ollama doesn’t contain curl. I haven’t yet figured out a good way to handle that. If you have suggestions I would love ideas

Reply:
I see, but we could at least allow users to plug their own commands in there, maybe they want to use Llama.cpp directly. I'm not sure if they're already doing this. Maybe the previous default container run command also works, we could try it out.

Reply:
I guess we could always restore original functionality and only support context from one chat to another if curl is available. I’ll try to play with that idea in the next few days

Reply:
Okay, that's also something for a follow up PR